### ISSUE TITLE
Inconsistency in mixed value/undefined array key offset detection for variables and array assignments

### ISSUE THREAD

**Issue by zerkms** [vimeo/psalm#11056]
```php
<?php
declare(strict_types=1);

function callme(array $arr): void
{
    $v = $arr['f1'];
    
    $v2 = [$arr['f1']];
    
    /** @psalm-suppress ForbiddenCode */
    var_dump($v, $v2);
}
```

https://psalm.dev/r/69ee9ca164

As you see line 8 does not trigger any warning.

Is it a deliberate limitation or was simply overlooked?

**Comment by zerkms** [comment]
Okay, I didn't have `ensureArrayStringOffsetsExist` enabled in this particular project, sorry.
